astroid/brain/brain_builtin_inference.py

```diff
@@ -481,6 +481,79 @@ def infer_slice(node, context=None):
     return slice_node
 
 
+def infer_open(node, context=None):  # pylint: disable=unused-argument
+    """Understand `open` (and `file` in Python 2) calls."""
+
+    if six.PY2:
+        code, module_name = _get_py2_open_call_model(node)
+    else:
+        code, module_name = _get_py3_open_call_model(node)
+
+    node = extract_node(code, module_name)
```

Comment:
Currently, for each inference of "open", mock is recreated each time. It may (and should) be created only once, and on each inference it should just create a Instance node.